test/integration/clustertrustbundles/field_selector_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustertrustbundles
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"testing"
+
+	certsv1alpha1 "k8s.io/api/certificates/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	kubeapiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+func TestCTBSignerNameFieldSelector(t *testing.T) {
+	ctx := context.Background()
+
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--feature-gates=ClusterTrustBundle=true"}, framework.SharedEtcd())
+	defer server.TearDownFn()
+
+	client := kubernetes.NewForConfigOrDie(server.ClientConfig)
+
+	bundle1 := &certsv1alpha1.ClusterTrustBundle{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo-com--bar-v1",
+		},
+		Spec: certsv1alpha1.ClusterTrustBundleSpec{
+			SignerName: "foo.com/bar",
+			TrustAnchors: mustMakePEMBlock("CERTIFICATE", nil, mustMakeCertificate(t, &x509.Certificate{
+				SerialNumber: big.NewInt(0),
+				Subject: pkix.Name{
+					CommonName: "root1",
+				},
+				IsCA:                  true,
+				BasicConstraintsValid: true,
+			})),
+		},
+	}
+	if _, err := client.CertificatesV1alpha1().ClusterTrustBundles().Create(ctx, bundle1, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Error while creating bundle1: %v", err)
+	}
+
+	bundle2 := &certsv1alpha1.ClusterTrustBundle{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo-com--bar-v2",
+		},
+		Spec: certsv1alpha1.ClusterTrustBundleSpec{
+			SignerName: "foo.com/bar",
+			TrustAnchors: mustMakePEMBlock("CERTIFICATE", nil, mustMakeCertificate(t, &x509.Certificate{
+				SerialNumber: big.NewInt(0),
+				Subject: pkix.Name{
+					CommonName: "root2",
+				},
+				IsCA:                  true,
+				BasicConstraintsValid: true,
+			})),
+		},
+	}
+	if _, err := client.CertificatesV1alpha1().ClusterTrustBundles().Create(ctx, bundle2, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Error while creating bundle2: %v", err)
+	}
+
+	bundle3 := &certsv1alpha1.ClusterTrustBundle{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "baz-com--bar-v1",
+		},
+		Spec: certsv1alpha1.ClusterTrustBundleSpec{
+			SignerName: "baz.com/bar",
+			TrustAnchors: mustMakePEMBlock("CERTIFICATE", nil, mustMakeCertificate(t, &x509.Certificate{
```

Comment:
ERROR(darwin/arm64): /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/clustertrustbundles/field_selector_test.go:87:4: unknown field TrustAnchors in struct literal of type "k8s.io/kubernetes/vendor/k8s.io/api/certificates/v1alpha1".ClusterTrustBundleSpec

Reply:
Done.